types/dockerode/index.d.ts

```diff
@@ -2018,36 +2023,29 @@ declare class Dockerode {
         image: string,
         cmd: string[],
         outputStream: NodeJS.WritableStream | NodeJS.WritableStream[],
-        createOptions: {},
-        startOptions: {},
-        callback: Callback<any>,
-    ): events.EventEmitter;
-    run(
-        image: string,
-        cmd: string[],
-        outputStream: NodeJS.WritableStream | NodeJS.WritableStream[],
-        startOptions: {},
+        createOptions: Dockerode.ContainerCreateOptions,
+        startOptions: Dockerode.ContainerStartOptions,
         callback: Callback<any>,
     ): events.EventEmitter;
     run(
         image: string,
         cmd: string[],
         outputStream: NodeJS.WritableStream | NodeJS.WritableStream[],
+        createOptions: Dockerode.ContainerCreateOptions,
         callback: Callback<any>,
     ): events.EventEmitter;
     run(
         image: string,
         cmd: string[],
         outputStream: NodeJS.WritableStream | NodeJS.WritableStream[],
-        createOptions: {},
         callback: Callback<any>,
     ): events.EventEmitter;
```

Comment:
For the above, I:
1. removed the function definition that included `startOptions` and not `createOptions`
2. Reordered the functions so it was in order of most specified to least specified

For (1), [looking at the code](https://github.com/apocas/dockerode/blob/bb88194ff250c4f5b9a0eee2688c5dcc5f623ef9/lib/docker.js#L1487-L1494) it doesn't look to differentiate are you passing `createOptions` or `startOptions` if you only pass one object, rather it just assumes its `createOptions`. By adding actual types here, having both lead to a TS error in my example.